core/create/src/prompts/oidc.ts

```diff
@@ -0,0 +1,274 @@
+// to authenticate deployments previously we can copy the permissions that user
+// had so that we don't have to do any guesswork about what AWS resources are
+// being touched.
+const getPreviousIAMPermissions = async (
+  accessKeyId: string,
+  secretAccessKey: string,
+  serviceName: string
+) => {
+  const clientOptions = {
+    credentials: {
+      accessKeyId,
+      secretAccessKey
+    },
+    region: 'eu-west-1'
+  }
+
+  // Get the AWS account ID via the API. We could use a map of AWS account
+  // names to IDs but I'm not sure it's a good idea to expose those in a public
+  // repo.
+  const stsClient = new STSClient(clientOptions)
+  const getCallerIdentityCommand = new GetCallerIdentityCommand({})
+  const { Account: accountId } = await stsClient.send(getCallerIdentityCommand)
+  if (!accountId) {
+    return
+  }
+
+  const iamClient = new IAMClient(clientOptions)
+  const getPolicyCommand = new GetPolicyCommand({
+    PolicyArn: `arn:aws:iam::${accountId}:policy/FTDeployPolicyFor_${serviceName}`
+  })
+  const { Policy: policy } = await iamClient.send(getPolicyCommand)
+  if (!policy) {
+    return
+  }
+
+  const getPolicyVersionCommand = new GetPolicyVersionCommand({
+    PolicyArn: policy.Arn,
+    VersionId: policy.DefaultVersionId
+  })
+  const policyVersion = await iamClient.send(getPolicyVersionCommand)
+
+  const policyDocument = policyVersion.PolicyVersion?.Document
+  if (policyDocument) {
+    return JSON.parse(decodeURIComponent(policyDocument))
+  }
+}
+
+export default async function oidcPrompt(): Promise<boolean> {
+  let cancelled = false
+  // support an early return if the user interrupts one of the prompts
+  const onCancel = () => {
+    cancelled = true
+  }
+  const { confirm } = await prompt({
+    name: 'confirm',
+    type: 'confirm',
+    message: 'Would you like OIDC authentication to be set up automatically for your Serverless app?'
+  })
+  if (!confirm || cancelled) {
+    return cancelled
+  }
+
+  const cloudformationTemplateRaw = await fs.readFile(
+    path.resolve(__dirname, '../../files/oidc-cloudformation.yml'),
+    'utf8'
+  )
+  const cloudformationTemplate = YAML.parseDocument(cloudformationTemplateRaw)
+
+  const { teamName, awsAccount } = await prompt(
+    [
+      {
+        name: 'teamName',
+        type: 'text',
+        message:
+          'What is the name of your team? This will be used to generate the path to store your configuration in the aws-composer repository.'
+      },
+      {
+        name: 'awsAccount',
+        type: 'select',
+        choices: ['infra-prod', 'next-prod'].map((account) => ({ title: account, value: account })),
+        message: 'Please select which FT AWS account you want your project to deploy to'
+      }
+    ],
+    { onCancel }
+  )
+  if (cancelled) {
+    return true
+  }
+
+  // access keys are pulled from Tool Kit's development vault
+  const vaultEnvVars = new VaultEnvVars(winstonLogger, {
+    environment: 'development',
```

Comment:
**suggestion:** We should set this in the production folder if we are expecting teams to rely on this. IMO the development folder suggests that if we delete a value from it, it will not effect people outside the code-owning team.